src/app/core/services/phase.service.ts

```diff
@@ -76,9 +76,22 @@ export class PhaseService {
    * If on Issue Dashboard, add previously visited repositories to otherRepos.
    * @param repo New current repository
    */
-  changeCurrentRepository(repo: Repo): void {
+  async changeCurrentRepository(repo: Repo) {
+    if (!repo) {
+      throw new Error('Invalid repo name. Please provide repo name in the format Org/Repo.');
+    }
+
     this.logger.info(`PhaseService: Changing current repository to '${repo}'`);
 
+    const isValidRepository = await this.githubService
+      .isRepositoryPresent(repo.owner, repo.name)
+      .toPromise()
+      .then((isValidRepository) => isValidRepository);
+
+    if (!isValidRepository) {
+      throw new Error('Invalid repo. Please check your organisation and repo name.');
+    }
+
```

Comment:
invoke `ErrorHandlingService` here to raise the error snackbar instead of throwing an `Error` to the calling function?

Reply:
I think this increases coupling between 2 unrelated parts of the programme (`PhaseService` is a "backend" service, whereas `ErrorHandlingService` is a "frontend" service). IMO it makes more sense to throw here and catch in the component instead, since `PhaseService` should not have the responsibility of initiating the display of the Error.